cumulusci/tasks/salesforce/tests/test_network_member_group.py

```diff
@@ -0,0 +1,594 @@
+
+        task.logger = Mock()
+        task.logger.info = Mock()
+
+        parent_ids_by_name = Mock()
+        parent_ids_by_name.items = Mock(
+            return_value=[("Name_0", "Id_0"), ("Name_1", None), ("Name_2", "Id_2")]
+        )
+
+        task._get_parent_ids_by_name = Mock(return_value=parent_ids_by_name)
+
+        task._create_network_member_group = Mock()
+
+        expected_create_network_member_group_calls = []
+        for parent_name, parent_id in parent_ids_by_name.items.return_value:
+            expected_create_network_member_group_calls.append(
+                call(sobject_type, parent_name, parent_id)
+            )
+
+        # Execute the test.
+        task._process_parent(sobject_type, record_names)
+
+        # Assert scenario execute as expected.
+        task.logger.info.assert_called_once_with(f"    {sobject_type}:")
+
+        task._get_parent_ids_by_name.assert_called_once_with(sobject_type, record_names)
+
+        parent_ids_by_name.items.assert_called_once_with()
+
+        task._create_network_member_group.assert_has_calls(
+            expected_create_network_member_group_calls
+        )
+
+    def test_create_network_member_group__parent_not_found_in_query(self):
+        network_name = "network_name"
+        sobject_type = "sobject_type"
+        parent_name = "parent_name"
+        parent_id = None
+
+        self.assertFalse(parent_id)
+
+        task = create_task(CreateNetworkMemberGroups, {"network_name": network_name})
+
+        task._parent_ids = set()
+
+        task._network_id = "network_id"
+
+        task.logger = Mock()
+        task.logger.info = Mock()
+        task.logger.warn = Mock()
+
+        task.sf = Mock()
+        task.sf.NetworkMemberGroup = Mock()
+
+        insert_response = Mock()
+        insert_response.get = Mock()
+        task.sf.NetworkMemberGroup.create = Mock(insert_response)
+
+        # Execute the test.
+        with self.assertRaises(CumulusCIException) as context:
+            task._create_network_member_group(sobject_type, parent_name, parent_id)
+
+        # Assert scenario execute as expected.
+        self.assertEqual(
+            f'No {sobject_type} record found with Name "{parent_name}"',
+            context.exception.args[0],
+        )
+
+        task.logger.info.assert_not_called()
+
+        task.logger.warn.assert_not_called()
+
+        task.sf.NetworkMemberGroup.create.assert_not_called()
+
+    def test_create_network_member_group__parent_already_exists(self):
+        network_name = "network_name"
+        sobject_type = "sobject_type"
+        parent_name = "parent_name"
+        parent_id = "parent_id"
+
+        self.assertTrue(parent_id)
+
+        task = create_task(CreateNetworkMemberGroups, {"network_name": network_name})
+
+        task._parent_ids = set()
+        task._parent_ids.add(parent_id)
+        self.assertTrue(parent_id in task._parent_ids)
```

Comment:
This assertion is testing the Python standard library, which is not needed.

Reply:
Agreed - thank you for the guidance. Learning a lot updating this older test.